be/src/storage/rowset/rowset.cpp

```diff
@@ -454,9 +451,6 @@ StatusOr<std::vector<vectorized::ChunkIteratorPtr>> Rowset::get_segment_iterator
             continue;
         }
         auto res = seg_ptr->new_iterator(schema, seg_options);
-        if (res.status().is_end_of_file()) {
```

Comment:
I think you can't delete this check. because when segment data is filtered by zonemap, it will return end_of_file.

Reply:
Returns `EndOfFile` if |read_options| has predicate, obviously the caller has not set this predicate, it will never encounter this end of file branch, and the SegmentReadOptions instance is not passed from caller but an instance in the internal of callee without setting the option predicates_for_zone_map

https://github.com/StarRocks/starrocks/blob/843ad16b76021e56ba22bfa602775673000c8de3/be/src/storage/rowset/segment.cpp#L193-L202

https://github.com/StarRocks/starrocks/blob/843ad16b76021e56ba22bfa602775673000c8de3/be/src/storage/rowset/rowset.cpp#L439-L446